jmix-flowui/flowui/src/main/java/io/jmix/flowui/xml/layout/loader/component/AbstractGridLoader.java

```diff
@@ -362,6 +383,157 @@ protected Optional<Renderer> loadRenderer(Element columnElement, @Nullable MetaP
+        Optional<Boolean> filterable = loadBoolean(element, "filterable");
+
+        if (filterableColumns && filterable.isEmpty()
+                || filterable.isPresent() && filterable.get()) {
+            if (dataLoader == null) {
+                throw new GuiDevelopmentException(resultComponent.getClass().getSimpleName() +
+                        " with a filterable column must have a DataLoader",
+                        context, "Component ID", resultComponent.getId());
+            }
+
+            setFilterComponent(column, dataLoader, metaClass, property);
+        }
+    }
+
+    protected void setFilterComponent(Column<?> column, DataLoader dataLoader,
+                                      MetaClass metaClass, String property) {
+        PropertyFilter<?> propertyFilter = createPropertyFilter(dataLoader, metaClass, property);
+        Component filterButton = createFilterButton(propertyFilter);
+
+        Component headerComponent = createHeaderComponent(column.getHeaderText(), filterButton);
+        column.setHeader(headerComponent);
+    }
+
+    protected PropertyFilter<?> createPropertyFilter(DataLoader dataLoader, MetaClass metaClass, String property) {
+        PropertyFilter<?> propertyFilter = getUiComponents().create(PropertyFilter.class);
+
+        propertyFilter.setDataLoader(dataLoader);
+        propertyFilter.setProperty(property);
+        propertyFilter.setAutoApply(false);
+
+        propertyFilter.setOperation(getPropertyFilterSupport().getDefaultOperation(metaClass, property));
+        propertyFilter.setOperationEditable(true);
+        propertyFilter.setParameterName(PropertyConditionUtils.generateParameterName(property));
+        propertyFilter.setWidthFull();
+
+        return propertyFilter;
+    }
+
+    protected Component createFilterButton(PropertyFilter<?> propertyFilter) {
+        JmixButton filterButton = getUiComponents().create(JmixButton.class);
+        filterButton.addThemeVariants(ButtonVariant.LUMO_TERTIARY_INLINE, ButtonVariant.LUMO_ICON);
+        filterButton.setIcon(VaadinIcon.FILTER.create());
+        filterButton.setClassName(LumoUtility.TextColor.TERTIARY);
+        filterButton.getElement().setAttribute(ATTRIBUTE_JMIX_ROLE_NAME, COLUMN_FILTER_BUTTON_ROLE);
+
+        // Workaround (waiting for overlay component),
+        // when device is small - standard dialog is used
+        Dialog overlay = createOverlay(propertyFilter, filterButton);
+
+        filterButton.addClickListener(__ -> {
+            overlay.open();
+
+            if (!isSmallDevice()) {
+                overlay.getElement().executeJs(getOverlayPositionExpression(), overlay, filterButton);
+            }
+        });
+
+        return filterButton;
+    }
+
+    @SuppressWarnings({"rawtypes"})
+    protected Dialog createOverlay(PropertyFilter propertyFilter, JmixButton filterButton) {
+        Dialog dialog = new Dialog(propertyFilter);
+        dialog.addClassName(COLUMN_FILTER_DIALOG_CLASSNAME);
```

Comment:
what if we set `themeName`?

Reply:
No)

Reply:
![image](https://github.com/jmix-framework/jmix/assets/6077149/b65e15c3-f270-4474-8ce4-4bfdecbd95c1)